### ISSUE TITLE
MDI command S4000 turns the spindle ON

### ISSUE THREAD

**Issue by zmrdko** [LinuxCNC/linuxcnc#2781]
## Here are the steps I follow to reproduce the issue:

 1. estop off
 2. turn machine on
 3. home all
 4. issue S4000 MDI command

## This is what I expected to happen:

S4000 command only changes preset spindle speed

## This is what happened instead:

S4000 command makes spindle FWD button appear as ON
(**halui.spindle.0.runs-forward** is switched to ON state, but **spindle.0.forward** is OFF)

## It worked properly before this:
a friend of mine declared, that in 2.7 it does not happen

## Information about my hardware and software:

 * I am using this Linux distribution and version (often, shown by `lsb_release -a`): Debian 12 Bookworm
 * I am using this kernel version (shown by `uname -a`): Linux linuxcnc 6.1.0-15-rt-amd64 #1 SMP PREEMPT_RT Debian 6.1.66-1 (2023-12-09) x86_64 GNU/Linux
 * I am running ...
   * [X] A binary version from linuxcnc.org (including buildbot.linuxcnc.org)
   * [ ] A binary I built myself
   * [ ] A binary version from some other source besides linuxcnc.org
 * I am using this LinuxCNC version (shown in package manager or, for git versions, `scripts/get-version-from-git`): 2.9.1
 * I am using this user interface (GUI) (e.g., AXIS, Touchy, gmoccapy, etc): probe_basic/axis
 * I am using this interface hardware vendor and chipset (e.g., parallel port, ethernet port, FPGA card): Mesa 7i95t

https://github.com/LinuxCNC/linuxcnc/assets/56800227/3078db1c-e95d-48aa-a550-269483afb872

**Comment by zmrdko** [comment]
can you please double check?
here is video of stock axis sim config and halshow spindle pins:
https://github.com/LinuxCNC/linuxcnc/assets/56800227/ef608f73-7007-4921-b8bf-41894842d172

You are right, that it does not affect spindle.0.froward pins, It affects only halui.spindle.0.forward pin.

**Comment by andypugh** [comment]
I think you mean the halui.spindle.0.runs-forward pin? 
Does it set the runs-reverse pin if you enter a negative speed? 

Are you prevented from pressing the spindle-start pin (on the basis that it is already pressed)

**Comment by c-morley** [comment]
This sounds like a bug just fixed..try the latest code in 2.9

**Comment by zmrdko** [comment]
> This sounds like a bug just fixed..try the latest code in 2.9

im using this apt source:
```
deb [arch=amd64,arm64 signed-by=/usr/share/keyrings/linuxcnc.gpg] https://www.linuxcnc.org/ bookworm base 2.9-uspace
deb-src [arch=amd64,arm64 signed-by=/usr/share/keyrings/linuxcnc.gpg] https://www.linuxcnc.org/ bookworm base 2.9-uspace
```
when this repository gets updated with bugfix you mentioned?

**Comment by zmrdko** [comment]
If I understand correctly, it will be fixed in version 2.10

**Comment by c-morley** [comment]
It should be fixed in 2.9.2 coming out soon ish.
I always compile the code so can't advise how to try earlier. Usually the buildbot builds the currentest code.

**Comment by zmrdko** [comment]
Thanks!
